main.tf

```diff
@@ -29,6 +29,23 @@ resource "azurerm_user_assigned_identity" "aks" {
   tags                = var.tags
 }
 
+data "azurerm_user_assigned_identity" "cluster_user_defined_identity" {
+  count = length(var.managed_identities.user_assigned_resource_ids) > 0 ? length(var.managed_identities.user_assigned_resource_ids) : 0
+
+  # /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/acceptanceTestResourceGroup1/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testIdentity
+  # name is obtained from the above string which is the user assigned resource id - reference https://github.com/Azure/terraform-azurerm-aks/blob/decb533e2785f965673698b0ac9949faca963f68/role_assignments.tf#L11
+  name                = split("/", tolist(var.managed_identities.user_assigned_resource_ids)[count.index])[8]
+  resource_group_name = split("/", tolist(var.managed_identities.user_assigned_resource_ids)[count.index])[4]
+}
+
+resource "azurerm_role_assignment" "network_contributor_on_subnet" {
+  // Use the principal_id from the user assigned identity if it exists, otherwise use the principal_id from the AKS cluster
+  // reference https://github.com/Azure/terraform-azurerm-aks/blob/decb533e2785f965673698b0ac9949faca963f68/role_assignments.tf#L27
+  principal_id         = coalesce(try(data.azurerm_user_assigned_identity.cluster_user_defined_identity[0].principal_id, azurerm_kubernetes_cluster.this.identity[0].principal_id))
```

Comment:
without and with availability zones examples use ```data.azurerm_user_assigned_identity.cluster_user_defined_identity[0].principal_id```
whilst azurerm_kubernetes_cluster.this.identity[0].principal_id) uses the generated identity